pyproject.toml

```diff
@@ -0,0 +1,114 @@
+[project]
+name = "dbt-common"
+version = "0.0.1"
+description = "The shared common utilities that dbt-core and adapter implementations use"
+readme = "README.md"
+requires-python = ">=3.8"
+license = "Apache-2.0"
+keywords = []
+authors = [
+  { name = "dbt Labs", email = "info@dbtlabs.com" },
+]
+classifiers = [
+  "Development Status :: 4 - Beta",
+  "Programming Language :: Python",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: Implementation :: CPython",
+  "Programming Language :: Python :: Implementation :: PyPy",
+]
+dependencies = [
+  "agate~=1.7.0",
+  "colorama>=0.3.9,<0.5",
+  "isodate>=0.6,<0.7",
+  "jsonschema~=4.0",
+  "Jinja2~=3.0",
+  "mashumaro[msgpack]~=3.9",
+  "pathspec>=0.9,<0.12", # TODO: I'm not sure this is needed.  check search.py?
+  "protobuf>=4.0.0",
+  "python-dateutil~=2.0",
+  "requests<3.0.0",
+  "typing-extensions~=4.4",
+]
+
+[build-system]
+requires = ["hatchling"]
+build-backend = "hatchling.build"
+
+[tool.hatch.build.targets.sdist]
+exclude = [
+  "/.github",
+  "/.changes",
+  ".changie.yaml",
+  ".gitignore",
+  ".pre-commit-config.yaml",
+  "CONTRIBUTING.md",
+  "MAKEFILE",
+  "/tests",
+]
+
+[tool.hatch.build.targets.wheel]
+packages = ["dbt_common"]
+
+[tool.hatch.envs.dev-env.scripts]
+all = ["pre-commit run --all-files"]
+
+[tool.hatch.envs.dev-env]
+description = "Env for running development commands like pytest / pre-commit"
```

Comment:
❤️ no more dev-requirements.txt? Yasss

Reply:
I think `hatch` lets you put these in the default environment as well. I definitely could be wrong as I'm just learning it. But that would let folks (especially external contributors) use `pip` if they really wanted to.

Reply:
One more comment, I think it could be worth creating a `test-env` environment for CI.

Reply:
I'm going to do more hatch specific tweaks in another PR where the adr for hatch will be created.

Reply:
[It's gone!](https://github.com/dbt-labs/dbt-common/pull/13/commits/beeb2f2f771a2799160c540577c8aebbdd6668ce)